plugin/kprom/kprom.go

```diff
@@ -298,3 +298,21 @@ func (m *Metrics) OnFetchBatchRead(meta kgo.BrokerMetadata, topic string, _ int3
 	node := strnode(meta.NodeID)
 	m.fetchBytes.WithLabelValues(node, topic).Add(float64(fbm.UncompressedBytes))
 }
+
+// Unregister the metrics if a registry was provided or created. Returns true if all metrics were unregistered.
+func (m *Metrics) Unregister() bool {
+	if m.cfg.reg == nil {
+		return false
+	}
+
+	a := m.cfg.reg.Unregister(m.connects)
+	b := m.cfg.reg.Unregister(m.connectErrs)
+	c := m.cfg.reg.Unregister(m.disconnects)
+	d := m.cfg.reg.Unregister(m.writeErrs)
+	e := m.cfg.reg.Unregister(m.writeBytes)
+	f := m.cfg.reg.Unregister(m.readErrs)
+	g := m.cfg.reg.Unregister(m.readBytes)
+	h := m.cfg.reg.Unregister(m.produceBytes)
+	i := m.cfg.reg.Unregister(m.fetchBytes)
```

Comment:
More future proof might be something like,
```go
var failed bool
for _, unreg := range []prometheus.Collector {
    ....
} {
    failed = failed || m.cfg.reg.Unregister(unreg)
}
return failed
```
but this bit is fine with me for now